modules/paraflare/functions/PARAFL/fn_flareFire.sqf

```diff
@@ -1,52 +1,48 @@
-
-    [_flare, "tin_flareShot"] remoteExec ["say3D"];
-
-    [{
-      params ["_flare", "_magColor"];
-
-      private _flareLight = "#lightpoint" createVehicle position _flare;
-      _flareLight lightAttachObject [_flare, [0, 0, 0]];
-
-      ["paraflare_evt_flareLight", [_flareLight, _magColor]] call CBA_fnc_globalEvent;
-
-      [{(getPosATL (_this select 0) select 2) < 1}, {
-        if (alive (_this select 0)) then {deleteVehicle (_this select 0)};
-        if (alive (_this select 1)) then {deleteVehicle (_this select 1)};
-      }, [_flare,_flareLight], 40, {
-        if (alive (_this select 0)) then {deleteVehicle (_this select 0)};
-        if (alive (_this select 1)) then {deleteVehicle (_this select 1)};
-      }] call CBA_fnc_waitUntilAndExecute;
-    }, [_flare, _magColor], 2.6] call CBA_fnc_waitAndExecute;
-  }, [_ammoType, _pos, _dir, _magColor], 0.95] call CBA_fnc_waitAndExecute;
-}, [_ammoType, _pos, _dir, _magColor], 1] call CBA_fnc_waitAndExecute;
+#include "script_component.hpp"
+
+params["_magType"];
+
+private _pos = player modelToWorld [0, 1, 0];
+_pos = [_pos select 0, _pos select 1, (_pos select 2)+1.5];
+private _dir = getDir player;
+private _ammoType = "",
+
+switch (_magType) do {
+	// ace throwable flares
+	case "ACE_HandFlare_White":{_ammoType = getText (configFile >> "CfgMagazines" >> "vn_m127_mag" >> "ammo");};
+	case "ACE_HandFlare_Green":{_ammoType = getText (configFile >> "CfgMagazines" >> "vn_m128_mag" >> "ammo");};
+	case "ACE_HandFlare_Yellow":{_ammoType = getText (configFile >> "CfgMagazines" >> "vn_m127_mag" >> "ammo");};
+	case "ACE_HandFlare_Red":{_ammoType = getText (configFile >> "CfgMagazines" >> "vn_m129_mag" >> "ammo");};
+	
+	// karts flaregun ammo
+	case "FlareWhite_F":{_ammoType = getText (configFile >> "CfgMagazines" >> "vn_m127_mag" >> "ammo");};
+	case "FlareGreen_F":{_ammoType = getText (configFile >> "CfgMagazines" >> "vn_m128_mag" >> "ammo");};
+	case "FlareYellow_F":{_ammoType = getText (configFile >> "CfgMagazines" >> "vn_m127_mag" >> "ammo");};
+	case "FlareRed_F":{_ammoType = getText (configFile >> "CfgMagazines" >> "vn_m129_mag" >> "ammo");};
+	
+	// SOG dispoable Flarelauncher Ammo
+	case "vn_m127_mag":{_ammoType = getText (configFile >> "CfgMagazines" >> "vn_m127_mag" >> "ammo");};
+	case "vn_m128_mag":{_ammoType = getText (configFile >> "CfgMagazines" >> "vn_m128_mag" >> "ammo");};
+	case "vn_m129_mag":{_ammoType = getText (configFile >> "CfgMagazines" >> "vn_m129_mag" >> "ammo");};
+};
+
+player removeMagazine _magType;
+
+[player, "Gear"] remoteExec ["playActionNow"];
+
+[{
+  Params ["_ammoType", "_pos", "_dir"];
+
+  [player, "tin_flareFire"] remoteExec ["say3D"];
+
+  [{
+	params ["_ammoType", "_pos", "_dir"];
+
+    private _flare = _ammoType createVehicle _pos;
+
+	_flare setVectorDirAndUp [[sin _dir * cos  GVAR(flareAngle), cos _dir * cos  GVAR(flareAngle), sin  GVAR(flareAngle)], [0,0,1]];
+
+    [_flare, "tin_flareShot"] remoteExec ["say3D"];

```

Comment:
Where does this sound come from, is it already in the framework?

Reply:
Alreayd in the framework/Paraflare module. i did not add or remove anything to do with sounds (new sounds mentioned in changes are just apart of the new projectile from SOG)